test/Microsoft.ML.Tests/Transformers/TextFeaturizerTests.cs

```diff
@@ -88,5 +88,114 @@ public void TextTokenizationWorkout()
             CheckEquality("Text", "tokenized.tsv");
             Done();
         }
+
+
+        [Fact]
+        public void TextNormalizationAndStopwordRemoverWorkout()
+        {
+            string sentimentDataPath = GetDataPath("wikipedia-detox-250-line-data.tsv");
+            var data = TextLoader.CreateReader(Env, ctx => (
+                    label: ctx.LoadBool(0),
+                    text: ctx.LoadText(1)), hasHeader: true)
+                .Read(new MultiFileSource(sentimentDataPath));
+
+            var invalidData = TextLoader.CreateReader(Env, ctx => (
+                    label: ctx.LoadBool(0),
+                    text: ctx.LoadFloat(1)), hasHeader: true)
+                .Read(new MultiFileSource(sentimentDataPath));
+
+            var est = new TextNormalizer(Env,"text")
+                .Append(new WordTokenizer(Env, "text", "words"))
+                .Append(new StopwordRemover(Env, "words", "words_without_stopwords"));
+            TestEstimatorCore(est, data.AsDynamic, invalidInput: invalidData.AsDynamic);
+
+            var outputPath = GetOutputPath("Text", "words_without_stopwords.tsv");
+            using (var ch = Env.Start("save"))
+            {
+                var saver = new TextSaver(Env, new TextSaver.Arguments { Silent = true });
+                IDataView savedData = TakeFilter.Create(Env, est.Fit(data.AsDynamic).Transform(data.AsDynamic), 4);
+                savedData = new ChooseColumnsTransform(Env, savedData, "text", "words_without_stopwords");
+
+                using (var fs = File.Create(outputPath))
+                    DataSaverUtils.SaveDataView(ch, saver, savedData, fs, keepHidden: true);
+            }
+
+            CheckEquality("Text", "words_without_stopwords.tsv");
+            Done();
+        }
+
+        [Fact]
+        public void WordBagWorkout()
+        {
+            string sentimentDataPath = GetDataPath("wikipedia-detox-250-line-data.tsv");
+            var data = TextLoader.CreateReader(Env, ctx => (
+                    label: ctx.LoadBool(0),
+                    text: ctx.LoadText(1)), hasHeader: true)
+                .Read(new MultiFileSource(sentimentDataPath));
+
+            var invalidData = TextLoader.CreateReader(Env, ctx => (
```

Comment:
Why is invalid data the same as the valid one?

Reply:
Its not valid. Here text is tried to be loaded as float.

---
In reply to: [219582043](https://github.com/dotnet/machinelearning/pull/953#discussion_r219582043) [](ancestors = 219582043)